test/e2e/tests/risingwave/tests.sh

```diff
@@ -109,10 +109,10 @@ function test::run::risingwave::multi_meta_failover_fencing() {
     return 1
   fi 
 
-  local old_leader_restarts
-  old_leader_restarts="$(k8s::kubectl::get pod "${meta_leader_names}" -o=jsonpath='{.items..status.containerStatuses..restartCount}')"
-  if ((old_leader_restarts <= meta_leaders_restarts)); then
-    logging::error "Leader did not restart"
+  local meta_leaders_restarts
+  meta_leaders_restarts="$(k8s::kubectl::get pod "${meta_leader_names}" -o=jsonpath='{.status.containerStatuses..restartCount}')"
```

Comment:
What if we have an issue in the election code or the operator and we get multiple leaders? `meta_leader_names` could be multiple names. 

Should we add an assertion that makes sure that `meta_leader_names` is only one item? Or should we just let it crash here?

Reply:
Agree. I think we should add an assertion before that. The problem before is the behaviour differs when the `meta_leader_names` contains only one single value -- there won't be an `items` field. And the subshell `$(...)` just pass without exiting the script (a "feature" of the bash, I really need to reconsider the PL for writing E2E tests 😭 ).

Reply:
Makes total sense. Do you want to add the assertion in this PR or in a follow-up?